### PR TITLE
Kv c4c

### PR DESCRIPTION
**Pull request by angelika233**
## Summary
Added key vault support in c4c tasks

## Importance
Introduces new functionality 

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] updates docstrings for any new functions or function arguments (if appropriate)
- [x] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

**Comment by angelika233** [comment]
@gwieloch

File: tests/integration/tasks/test_cloud_for_customers.py [hunk 1]
```diff
@@ -8,7 +8,7 @@ def test_c4c_to_df():
     c4c_to_df = C4CToDF()
     df = c4c_to_df.run(url=url, endpoint=endpoint)
     answer = df.head()
-
+    print(df)
```
Comment (lzuchowska) [inline tests/integration/tasks/test_cloud_for_customers.py:11]: Prints() should be deleted 🤖

File: viadot/tasks/cloud_for_customers.py [hunk 2]
```diff
@@ -128,12 +154,33 @@ def run(
...
+            credentials = json.loads(credentials_str)[env]
+
         cloud_for_customers = CloudForCustomers(
-            url=url, params=params, endpoint=endpoint, env=env, fields=fields
+            url=url,
+            params=params,
+            endpoint=endpoint,
+            env=env,
+            fields=fields,
+            credentials=credentials,
```
Comment (lzuchowska) [inline viadot/tasks/cloud_for_customers.py:183]: If credentials are None (for instance: test_cloud_for_customers:test_c4c_to_df()) this will fail 😭

Comment (lzuchowska) [reply viadot/tasks/cloud_for_customers.py:183]: When I run the tests, test_c4c_to_df() fails with an error:
```UnboundLocalError: local variable 'credentials' referenced before assignment```.

The test uses `C4CToDF` class and its run() method, without a `credentials_secret` parameter. This should be handled - right now, in run() method, CloudForCustomers() is being called on line 177, with the `credentials` parameter. Take a closer look on when the credentials parameter is being defined ✨ 😱

File: viadot/tasks/cloud_for_customers.py [hunk 3]
```diff
@@ -128,12 +157,34 @@ def run(
...
             if_empty (str, optional): What to do if query returns no data. Defaults to "warn".
+            credentials_secret (str, optional): The name of the Azure Key Vault secret containing a dictionary
+            with C4C credentials. Defaults to None.
+            vault_name (str, optional): The name of the vault from which to obtain the secret. Defaults to None.
 
         Returns:
             pd.DataFrame: The query result as a pandas DataFrame.
         """
+        if not credentials_secret:
+            # attempt to read a default for the service principal secret name
```
Comment (Trymzet) [inline viadot/tasks/cloud_for_customers.py:168]: this comment doesn't apply to this situation

**Pull request merged**